linera-indexer/src/indexer.rs

```diff
@@ -130,18 +109,44 @@ where
         if height < listener.start {
             return Ok(());
         };
-        let last_hash = match state.chains.get(&chain_id).await? {
-            None => None,
+        let latest_block = match state.chains.get(&chain_id).await? {
+            None => LatestBlock::StartHeight(listener.start),
             Some((last_hash, last_height)) => {
                 if last_hash == hash || last_height >= height {
                     return Ok(());
                 }
-                Some(last_hash)
+                LatestBlock::LatestHash(last_hash)
             }
         };
         info!("process {:?}: {:?} ({})", chain_id, hash, height);
-        self.process_rec(&listener.service, state, value, last_hash)
-            .await
+
+        let mut values = Vec::new();
+        let mut value = value.clone();
+        loop {
+            let Some(block) = value.inner().block() else {
+                break;
+            };
+            if !value.inner().is_confirmed() {
+                break;
+            };
```

Comment:
If we `break` here it will process the other values, which seems wrong, since this invalid value is one of their ancestors. I'd rather `return`.